bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -128,7 +128,12 @@ void ReplicaImp::registerMsgHandlers() {
 template <typename T>
 void ReplicaImp::messageHandler(MessageBase *msg) {
   T *trueTypeObj = new T(msg);
+  if (isCollectingState()) {
```

Comment:
nit:
thinking quickly but didn't spent more than a minute on it: In general, it is not a good idea to handle a message straight to ST to sniff, you didn't validate the message?